File: go/services/order/app/core/model/order.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type OrderStatus string

const (
	OrderPlaced     OrderStatus = "order_placed"
	OrderInProgress OrderStatus = "order_in_progress"
	OrderCanceled   OrderStatus = "order_canceled"
	OrderCompleted  OrderStatus = "order_completed"
)

type OrderId string

type Order struct {
	OrderId      OrderId
	CustomerId   uuid.UUID
	CreationDate time.Time
	Status       OrderStatus
	Workflow     string
	Items        []OrderItem
}

type OrderItem struct {
	OrderItemId  int
	Name         string
	CreationDate time.Time
}

File: rust/services/order/app/src/core/model/order.rs
use strum_macros::{Display, EnumString};
use time::OffsetDateTime;

use crate::core::service::order_id::OrderId;

#[derive(Debug, Display, EnumString)]
pub enum OrderStatus {
    OrderPlaced,
    // OrderInProgress,
    // OrderCancelled,
    // OrderCompleted,
}

pub struct Order {
    pub order_id: OrderId,
    pub creation_date: OffsetDateTime,
    pub status: OrderStatus,
    pub items: Vec<OrderItem>,
}

pub struct OrderItem {
    pub order_item_id: i32,
    pub name: String,
    pub creation_date: OffsetDateTime,
}
